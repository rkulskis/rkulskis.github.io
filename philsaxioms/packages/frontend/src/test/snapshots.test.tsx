import { describe, it, expect } from 'vitest';

describe('Snapshots', () => {
  it('should render', () => {
    expect(true).toBe(true);
  });
});