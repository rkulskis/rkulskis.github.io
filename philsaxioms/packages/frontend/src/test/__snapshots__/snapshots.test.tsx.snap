// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component Snapshots > renders ArgumentNode for different levels 1`] = `
<div
  animate="[object Object]"
  class="relative p-4 min-w-[240px] max-w-[320px] rounded-lg shadow-lg cursor-pointer transition-all duration-200 border-2 border-green-400 bg-green-50"
  initial="[object Object]"
  transition="[object Object]"
  whilehover="[object Object]"
  whiletap="[object Object]"
>
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    class="absolute -top-2 -left-2 px-2 py-1 rounded-full text-xs font-bold text-white bg-purple-500"
  >
    L
    2
  </div>
  <div
    class="absolute -top-2 -right-2 px-2 py-1 rounded-full text-xs font-medium text-white"
    style="background-color: rgb(239, 68, 68);"
  >
    Ethics
  </div>
  <div
    class="absolute top-2 left-2 flex items-center gap-1"
  >
    <svg
      class="lucide lucide-target w-3 h-3 text-green-500"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <circle
        cx="12"
        cy="12"
        r="6"
      />
      <circle
        cx="12"
        cy="12"
        r="2"
      />
    </svg>
  </div>
  <div
    class="space-y-3 mt-4"
  >
    <h3
      class="font-bold text-gray-900 leading-tight text-sm"
    >
      Level 2 Argument
    </h3>
    <p
      class="text-xs text-gray-600 leading-relaxed"
    >
      This is a test argument that builds on axioms
    </p>
    <div
      class="border-t border-gray-200 pt-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <svg
          class="lucide lucide-chevron-right w-4 h-4 text-blue-500 mt-0.5 flex-shrink-0"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
        <p
          class="text-xs font-medium text-blue-700 leading-tight"
        >
          Therefore, this conclusion follows logically
        </p>
      </div>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <span
        class="text-xs text-gray-500"
      >
        Strength:
      </span>
      <div
        class="flex-1 bg-gray-200 rounded-full h-1"
      >
        <div
          class="bg-green-500 h-1 rounded-full transition-all duration-300"
          style="width: 80%;"
        />
      </div>
      <span
        class="text-xs text-gray-500"
      >
        80
        %
      </span>
    </div>
    <div
      class="flex flex-wrap gap-1"
    >
      <span
        class="px-1 py-0.5 bg-gray-100 text-gray-600 text-xs rounded"
      >
        reasoning
      </span>
      <span
        class="px-1 py-0.5 bg-gray-100 text-gray-600 text-xs rounded"
      >
        logic
      </span>
    </div>
  </div>
</div>
`;

exports[`Component Snapshots > renders ArgumentNode in invalid state 1`] = `
<div
  animate="[object Object]"
  class="relative p-4 min-w-[240px] max-w-[320px] rounded-lg shadow-lg cursor-pointer transition-all duration-200 border-2 border-gray-300 bg-gray-50"
  initial="[object Object]"
  transition="[object Object]"
  whilehover="[object Object]"
  whiletap="[object Object]"
>
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    class="absolute -top-2 -left-2 px-2 py-1 rounded-full text-xs font-bold text-white bg-blue-500"
  >
    L
    1
  </div>
  <div
    class="absolute -top-2 -right-2 px-2 py-1 rounded-full text-xs font-medium text-white"
    style="background-color: rgb(239, 68, 68);"
  >
    Ethics
  </div>
  <div
    class="absolute top-2 left-2 flex items-center gap-1"
  >
    <svg
      class="lucide lucide-target w-3 h-3 text-gray-400"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <circle
        cx="12"
        cy="12"
        r="6"
      />
      <circle
        cx="12"
        cy="12"
        r="2"
      />
    </svg>
  </div>
  <div
    class="space-y-3 mt-4"
  >
    <h3
      class="font-bold text-gray-900 leading-tight text-sm"
    >
      Test Argument
    </h3>
    <p
      class="text-xs text-gray-600 leading-relaxed"
    >
      This is a test argument that builds on axioms
    </p>
    <div
      class="border-t border-gray-200 pt-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <svg
          class="lucide lucide-chevron-right w-4 h-4 text-blue-500 mt-0.5 flex-shrink-0"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
        <p
          class="text-xs font-medium text-blue-700 leading-tight"
        >
          Therefore, this conclusion follows logically
        </p>
      </div>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <span
        class="text-xs text-gray-500"
      >
        Strength:
      </span>
      <div
        class="flex-1 bg-gray-200 rounded-full h-1"
      >
        <div
          class="bg-green-500 h-1 rounded-full transition-all duration-300"
          style="width: 80%;"
        />
      </div>
      <span
        class="text-xs text-gray-500"
      >
        80
        %
      </span>
    </div>
    <div
      class="flex flex-wrap gap-1"
    >
      <span
        class="px-1 py-0.5 bg-gray-100 text-gray-600 text-xs rounded"
      >
        reasoning
      </span>
      <span
        class="px-1 py-0.5 bg-gray-100 text-gray-600 text-xs rounded"
      >
        logic
      </span>
    </div>
  </div>
</div>
`;

exports[`Component Snapshots > renders ArgumentNode in valid state 1`] = `
<div
  animate="[object Object]"
  class="relative p-4 min-w-[240px] max-w-[320px] rounded-lg shadow-lg cursor-pointer transition-all duration-200 border-2 border-green-400 bg-green-50"
  initial="[object Object]"
  transition="[object Object]"
  whilehover="[object Object]"
  whiletap="[object Object]"
>
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    class="absolute -top-2 -left-2 px-2 py-1 rounded-full text-xs font-bold text-white bg-blue-500"
  >
    L
    1
  </div>
  <div
    class="absolute -top-2 -right-2 px-2 py-1 rounded-full text-xs font-medium text-white"
    style="background-color: rgb(239, 68, 68);"
  >
    Ethics
  </div>
  <div
    class="absolute top-2 left-2 flex items-center gap-1"
  >
    <svg
      class="lucide lucide-target w-3 h-3 text-green-500"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <circle
        cx="12"
        cy="12"
        r="6"
      />
      <circle
        cx="12"
        cy="12"
        r="2"
      />
    </svg>
  </div>
  <div
    class="space-y-3 mt-4"
  >
    <h3
      class="font-bold text-gray-900 leading-tight text-sm"
    >
      Test Argument
    </h3>
    <p
      class="text-xs text-gray-600 leading-relaxed"
    >
      This is a test argument that builds on axioms
    </p>
    <div
      class="border-t border-gray-200 pt-2"
    >
      <div
        class="flex items-start gap-2"
      >
        <svg
          class="lucide lucide-chevron-right w-4 h-4 text-blue-500 mt-0.5 flex-shrink-0"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m9 18 6-6-6-6"
          />
        </svg>
        <p
          class="text-xs font-medium text-blue-700 leading-tight"
        >
          Therefore, this conclusion follows logically
        </p>
      </div>
    </div>
    <div
      class="flex items-center gap-2"
    >
      <span
        class="text-xs text-gray-500"
      >
        Strength:
      </span>
      <div
        class="flex-1 bg-gray-200 rounded-full h-1"
      >
        <div
          class="bg-green-500 h-1 rounded-full transition-all duration-300"
          style="width: 80%;"
        />
      </div>
      <span
        class="text-xs text-gray-500"
      >
        80
        %
      </span>
    </div>
    <div
      class="flex flex-wrap gap-1"
    >
      <span
        class="px-1 py-0.5 bg-gray-100 text-gray-600 text-xs rounded"
      >
        reasoning
      </span>
      <span
        class="px-1 py-0.5 bg-gray-100 text-gray-600 text-xs rounded"
      >
        logic
      </span>
    </div>
  </div>
</div>
`;

exports[`Component Snapshots > renders AxiomNode in accepted state 1`] = `
<div
  animate="[object Object]"
  class="axiom-node relative p-4 min-w-[200px] max-w-[280px] border-green-500 bg-green-50"
  initial="[object Object]"
  transition="[object Object]"
  whilehover="[object Object]"
  whiletap="[object Object]"
>
  <div
    class="lego-connector top"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector bottom"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector left"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector right"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    class="absolute -top-2 -right-2 px-2 py-1 rounded-full text-xs font-medium text-white"
    style="background-color: rgb(239, 68, 68);"
  >
    Ethics
  </div>
  <div
    class="space-y-2"
  >
    <h3
      class="font-semibold text-gray-900 leading-tight"
    >
      Test Axiom
    </h3>
    <p
      class="text-sm text-gray-600 leading-relaxed"
    >
      This is a test axiom for snapshot testing
    </p>
    <div
      class="flex flex-wrap gap-1 mt-2"
    >
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        test
      </span>
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        philosophy
      </span>
    </div>
  </div>
  <div
    class="absolute top-2 left-2 w-3 h-3 bg-green-500 rounded-full"
  />
</div>
`;

exports[`Component Snapshots > renders AxiomNode in default state 1`] = `
<div
  animate="[object Object]"
  class="axiom-node relative p-4 min-w-[200px] max-w-[280px] border-gray-300"
  initial="[object Object]"
  style="border-color: rgb(239, 68, 68);"
  transition="[object Object]"
  whilehover="[object Object]"
  whiletap="[object Object]"
>
  <div
    class="lego-connector top"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector bottom"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector left"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector right"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    class="absolute -top-2 -right-2 px-2 py-1 rounded-full text-xs font-medium text-white"
    style="background-color: rgb(239, 68, 68);"
  >
    Ethics
  </div>
  <div
    class="space-y-2"
  >
    <h3
      class="font-semibold text-gray-900 leading-tight"
    >
      Test Axiom
    </h3>
    <p
      class="text-sm text-gray-600 leading-relaxed"
    >
      This is a test axiom for snapshot testing
    </p>
    <div
      class="flex flex-wrap gap-1 mt-2"
    >
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        test
      </span>
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        philosophy
      </span>
    </div>
  </div>
</div>
`;

exports[`Component Snapshots > renders AxiomNode in rejected state 1`] = `
<div
  animate="[object Object]"
  class="axiom-node relative p-4 min-w-[200px] max-w-[280px] border-red-500 bg-red-50"
  initial="[object Object]"
  transition="[object Object]"
  whilehover="[object Object]"
  whiletap="[object Object]"
>
  <div
    class="lego-connector top"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector bottom"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector left"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector right"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    class="absolute -top-2 -right-2 px-2 py-1 rounded-full text-xs font-medium text-white"
    style="background-color: rgb(239, 68, 68);"
  >
    Ethics
  </div>
  <div
    class="space-y-2"
  >
    <h3
      class="font-semibold text-gray-900 leading-tight"
    >
      Test Axiom
    </h3>
    <p
      class="text-sm text-gray-600 leading-relaxed"
    >
      This is a test axiom for snapshot testing
    </p>
    <div
      class="flex flex-wrap gap-1 mt-2"
    >
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        test
      </span>
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        philosophy
      </span>
    </div>
  </div>
  <div
    class="absolute top-2 left-2 w-3 h-3 bg-red-500 rounded-full"
  />
</div>
`;

exports[`Component Snapshots > renders AxiomNode in selected state 1`] = `
<div
  animate="[object Object]"
  class="axiom-node relative p-4 min-w-[200px] max-w-[280px] border-gray-300 ring-4 ring-blue-400 ring-opacity-50"
  initial="[object Object]"
  style="border-color: rgb(239, 68, 68);"
  transition="[object Object]"
  whilehover="[object Object]"
  whiletap="[object Object]"
>
  <div
    class="lego-connector top"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector bottom"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector left"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    class="lego-connector right"
    style="background-color: rgb(239, 68, 68);"
    transition="[object Object]"
    whilehover="[object Object]"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    data-testid="handle"
  />
  <div
    class="absolute -top-2 -right-2 px-2 py-1 rounded-full text-xs font-medium text-white"
    style="background-color: rgb(239, 68, 68);"
  >
    Ethics
  </div>
  <div
    class="space-y-2"
  >
    <h3
      class="font-semibold text-gray-900 leading-tight"
    >
      Test Axiom
    </h3>
    <p
      class="text-sm text-gray-600 leading-relaxed"
    >
      This is a test axiom for snapshot testing
    </p>
    <div
      class="flex flex-wrap gap-1 mt-2"
    >
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        test
      </span>
      <span
        class="px-2 py-1 bg-gray-100 text-gray-600 text-xs rounded"
      >
        philosophy
      </span>
    </div>
  </div>
</div>
`;

exports[`Component Snapshots > renders Questionnaire component 1`] = `
<div
  class="min-h-screen bg-gradient-to-br from-blue-50 to-indigo-100 flex items-center justify-center p-4"
>
  <div
    class="max-w-2xl w-full"
  >
    <div
      class="text-center mb-8"
    >
      <h1
        class="text-3xl font-bold text-gray-900 mb-4"
      >
        PhilsAxioms
      </h1>
      <p
        class="text-gray-600 mb-4"
      >
        Build your philosophical framework by answering these foundational questions
      </p>
      <button
        class="text-indigo-600 hover:text-indigo-800 underline text-sm"
      >
        Skip questionnaire and explore all axioms →
      </button>
    </div>
    <div
      class="mb-8"
    >
      <div
        class="flex justify-between text-sm text-gray-600 mb-2"
      >
        <span>
          Question 
          1
           of 
          1
        </span>
        <span>
          100
          % complete
        </span>
      </div>
      <div
        class="w-full bg-gray-200 rounded-full h-2"
      >
        <div
          animate="[object Object]"
          class="bg-indigo-600 h-2 rounded-full"
          initial="[object Object]"
          transition="[object Object]"
        />
      </div>
    </div>
    <div>
      <div
        animate="[object Object]"
        class="bg-white rounded-xl shadow-xl p-8"
        exit="[object Object]"
        initial="[object Object]"
        transition="[object Object]"
      >
        <h2
          class="text-2xl font-bold text-gray-900 mb-4"
        >
          Do you believe there is an objective reality?
        </h2>
        <p
          class="text-gray-600 mb-8 leading-relaxed"
        >
          This is about whether reality exists independently
        </p>
        <div
          class="flex gap-4 mb-6"
        >
          <button
            class="flex-1 bg-green-500 hover:bg-green-600 text-white font-semibold py-4 px-6 rounded-lg flex items-center justify-center gap-2 transition-colors"
            whilehover="[object Object]"
            whiletap="[object Object]"
          >
            <svg
              class="lucide lucide-check w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 6 9 17l-5-5"
              />
            </svg>
            Yes, I accept this
          </button>
          <button
            class="flex-1 bg-red-500 hover:bg-red-600 text-white font-semibold py-4 px-6 rounded-lg flex items-center justify-center gap-2 transition-colors"
            whilehover="[object Object]"
            whiletap="[object Object]"
          >
            <svg
              class="lucide lucide-x w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 6 6 18"
              />
              <path
                d="m6 6 12 12"
              />
            </svg>
            No, I reject this
          </button>
        </div>
        <div
          class="flex justify-between"
        >
          <button
            class="flex items-center gap-2 text-gray-500 hover:text-gray-700 disabled:opacity-50 disabled:cursor-not-allowed"
            disabled=""
          >
            <svg
              class="lucide lucide-chevron-left w-4 h-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15 18-6-6 6-6"
              />
            </svg>
            Previous
          </button>
          <span
            class="text-gray-400"
          >
            Finish
            <svg
              class="lucide lucide-chevron-right w-4 h-4 inline ml-1"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
